workflow/controller/operator.go

```diff
@@ -1460,16 +1461,21 @@ func (woc *wfOperationCtx) executeTemplate(nodeName string, orgTmpl wfv1.Templat
 			node = lastChildNode
 		} else {
 			// Create a new child node and append it to the retry node.
-			nodeName = fmt.Sprintf("%s(%d)", retryNodeName, len(retryParentNode.Children))
+			childrenLength := strconv.Itoa(len(retryParentNode.Children))
+			nodeName = fmt.Sprintf("%s(%s)", retryNodeName, childrenLength)
 			woc.addChildNode(retryNodeName, nodeName)
 			node = nil
 
+			localParams := make(map[string]string)
 			// Change the `pod.name` variable to the new retry node name
 			if processedTmpl.IsPodType() {
-				processedTmpl, err = common.SubstituteParams(processedTmpl, map[string]string{}, map[string]string{common.LocalVarPodName: woc.wf.NodeID(nodeName)})
-				if err != nil {
-					return woc.initializeNodeOrMarkError(node, nodeName, wfv1.NodeTypeSkipped, templateScope, orgTmpl, opts.boundaryID, err), err
-				}
+				localParams[common.LocalVarPodName] = woc.wf.NodeID(nodeName)
+			}
+			// Inject the retryAttempt number
+			localParams[common.LocalVarPodRetryAttempt] = childrenLength
```

Comment:
Following from previous comment:

```suggestion
			localParams[common.LocalVarPodRetryAttempt] = fmt.Sprint(len(retryParentNode.Children))
```